src/wazuh_db/wdb_syscollector.c

```diff
@@ -1107,8 +1107,6 @@ int wdb_syscollector_processes_save2(wdb_t * wdb, const cJSON * attributes)
     const int tty = cJSON_GetObjectItem(attributes, "tty") ? cJSON_GetObjectItem(attributes, "tty")->valueint : 0;
     const int processor = cJSON_GetObjectItem(attributes, "processor") ? cJSON_GetObjectItem(attributes, "processor")->valueint : 0;
     const char * checksum = cJSON_GetStringValue(cJSON_GetObjectItem(attributes, "checksum"));
-    //this is needed to remove old entries in the db with a scan_id != 0: https://github.com/wazuh/wazuh/issues/7356
-    wdb_process_delete(wdb, scan_id);
```

Comment:
Shouldn't we need to remove the `wdb_process_delete` (declared in line 1051) since is not used anymore?

Reply:
it's being used by legacy agents in wdb_parser.c